File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    i = 1
    while True:
        y_h = i
        y_w = yellow / y_h

        if brown == (y_h * 2) + ((y_w + 2) * 2):
            answer = [round(y_w + 2), (y_h + 2)]
            break
        i += 1

    return answer

File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    let width = brown / 2 
    let height = 2
    let result = -1
    
    while (result !== 0) {
        width--
        height++
        let mul = width * height
        result = mul - brown - yellow
    }
    return [width, height]
}